pandas/_libs/tslibs/period.pyx

```diff
@@ -886,7 +886,10 @@ cdef int64_t get_time_nanos(int freq, int64_t unix_date, int64_t ordinal) nogil:
         # We must have freq == FR_HR
         factor = 10**9 * 3600
 
-    sub = ordinal - unix_date * 24 * 3600 * 10**9 / factor
+    # Fix issue #36430
+    nanos_in_day = 24 * 3600 * 10**9
```

Comment:
Can you add the appropriate suffixes to the constants here? This seems suspect that it would make a difference at all; wonder if the suffixes alone would fix

Reply:
Do you mean declaring this way: `cdef const nanos_in_day = 24 * 3600 * 10**9` ?
This does not compile.